pkg/apis/triggers/v1alpha1/event_listener_types.go

```diff
@@ -36,10 +37,18 @@ type EventListenerSpec struct {
 //
 // +k8s:deepcopy-gen=true
 type Trigger struct {
+	TriggerValidate *TriggerValidate   `json:"validate,omitempty"`
 	TriggerBinding  TriggerBindingRef  `json:"binding"`
 	TriggerTemplate TriggerTemplateRef `json:"template"`
 }
 
+// TriggerValidate represents the image to run taskrun for validating that trigger comes from the source which is desired
+type TriggerValidate struct {
+	TaskRef        *pipelinev1.TaskRef `json:"taskRef"`
```

Comment:
We should add some documentation around this. What Tasks are allowed e.g. they need to take a payload parameter and should not consume any resources etc.
 
In the future, we should validation for the Task to ensure it is a proper validation task (maybe instead of a Task we specify a specialized type which is a subset of a Task such as a `Validation` or `Filter`)